### ISSUE TITLE
[Bug]: `generateJSON` can return links `href` as number

### ISSUE THREAD

**Issue by ChidVanid** [ueberdosis/tiptap#4929]
### Which packages did you experience the bug in?

extension-link, html

### What Tiptap version are you using?

tiptap: 2.2.4, extension-link: 2.2.4, html: 2.2.4

### What’s the bug you are facing?

We have historically links like `<p><a href="123">Test</a></p>` in our data.
Before inserting this content in our tiptap editor we use `generateJSON` from the html package. The JSON this function returns is later used in the editors `setContent` command.
After `setContent` we get the following error: `HTMLAttributes.href.startsWith is not a function`.
After some inspection i could pinpoint it to the link extensions `renderHTML` function. The if statement there which checks if the href starts with `javascript:` fails since the href arrives in this function as a `Number` instead of a `String`.
For whatever reason `generateJSON` parses values like `123` as Numbers.

To fix this we extended the link extension and replaced the `renderHTML` function to handle actual numbers but I guess it would be the better way to fix this in the `generateJSON` function?

### What browser are you using?

Chrome

### Code example

https://codesandbox.io/p/sandbox/tender-rhodes-jhhpgg

### What did you expect to happen?

`href` to be parsed as String.

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [ ] Yes, I’m a sponsor. 💖

**Comment by denego** [comment]
Got the same issue, had to workaround like this:
```
import { mergeAttributes } from '@tiptap/core';
import LinkBase from '@tiptap/extension-link';

export const Link = LinkBase
  .extend({
    renderHTML({ HTMLAttributes }) {
      // force HTMLAttributes.href to be a string
      return ['a', mergeAttributes(this.options.HTMLAttributes, {
        ...HTMLAttributes, href: String(HTMLAttributes.href),
      }), 0];
    },
  });
```

**Comment by yurtsiv** [comment]
To prevent numbers from entering your doc in the first place, you can do the following:

```typescript
const Link = LinkBase.extend({
  addAttributes() {
    return {
      ...(this.parent?.() ?? {}),
      href: {
        default: null,
        parseHTML(element) {
          return element.getAttribute('href');
        },
      }
    }
  },
})
```

If `parseHTML` is not provided, TipTap will get the attribute for you and [convert it](https://github.com/ueberdosis/tiptap/blob/341b01d384995aa429593a0842b817516f4111f6/packages/core/src/helpers/injectExtensionAttributesToParseRule.ts#L32) to number/boolean if possible. It's a bug in the link extension.

**Comment by nperez0111** [comment]
This is resolved with 2.6.0
